### PR TITLE
Run notebook examples in CI

### PR DESCRIPTION
**Pull request by Saityi**
## Issue

Our notebook examples easily go out of date, and it's hard to notice when it happens without a regular CI checking them

## Fix

Convert notebook examples to Julia code and run them

Fix errors CI exposed:
- `Graph` was removed from `visualization.jl`, but was used by examples, especially older ones
- `ClOUD` -> `CLOUD`
- `Graph` -> `GraphCL`
- Remove `Pkg.activate('../../../')` in a causal loop example
- Fix a causal loop example to use vectors instead of tuples

## Notes

Includes commits from #131 and verifies removing `GraphViz` did not break any examples. @KevinDCarlson

Also addresses https://github.com/AlgebraicJulia/StockFlow.jl/issues/113#issuecomment-1861809454

## Misc

- Removed ipynb-checkpoints and add a more generalised pattern to exclude them
- Ignore vim & emacs temporary files
- Clean up a couple complaints from `Aqua.jl` around non-existent exports 
- Renamed `full_fledged_schema_examples/CaulsalLoopDiagrams` to `full_fledged_schema_examples/CausalLoopDiagrams` 

These tests can be run locally by running the helper script, `validate-notebook-examples.sh`, which does what the workflow does:
``` bash
$ validate-notebook-examples.sh
Starting to run notebooks ...
Checking: ["./jlexamples/full_fledged_schema_examples.COVID_full_model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.Pertussis Model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.SEIR_full_model_measles_chickenpox.ipynb.jl", "./jlexamples/full_fledged_schema_examples.SIR_example_function_structure_test_compostion_and_stratification.ipynb.jl", "./jlexamples/full_fledged_schema_examples.composed open population SIRV model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.convert_from_SEIR_stockFlowDiagram.ipynb.jl", "./jlexamples/full_fledged_schema_examples.create_a_causalLoopDiagram.ipynb.jl", "./jlexamples/full_fledged_schema_examples.curable sexually transmitted diseases model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.diabetes model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.diabetes_diagnose.ipynb.jl", "./jlexamples/full_fledged_schema_examples.sex_transmission_SIS.ipynb.jl", "./jlexamples/full_fledged_schema_examples_new.COVID_full_model.ipynb.jl", "./jlexamples/
OK: Any["./jlexamples/full_fledged_schema_examples.COVID_full_model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.Pertussis Model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.SEIR_full_model_measles_chickenpox.ipynb.jl", "./jlexamples/full_fledged_schema_examples.SIR_example_function_structure_test_compostion_and_stratification.ipynb.jl", "./jlexamples/full_fledged_schema_examples.composed open population SIRV model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.convert_from_SEIR_stockFlowDiagram.ipynb.jl", "./jlexamples/full_fledged_schema_examples.create_a_causalLoopDiagram.ipynb.jl", "./jlexamples/full_fledged_schema_examples.curable sexually transmitted diseases model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.diabetes model.ipynb.jl", "./jlexamples/full_fledged_schema_examples.diabetes_diagnose.ipynb.jl", "./jlexamples/full_fledged_schema_examples.sex_transmission_SIS.ipynb.jl", "./jlexamples/full_fledged_schema_examples_new.COVID_full_model.ipynb.jl", "./jlexamples/ful
```

## TODO

- In line with the comment in https://github.com/AlgebraicJulia/StockFlow.jl/pull/128#pullrequestreview-2236662771 we should consider renaming functions like `Graph` and what our 'display' API looks like going into a full initial 1.0
- Speed up notebook regression tests (e.g., parallelism in the test runner or performance improvements in the library). 
    - I tried a simple speed up using the `Threads.@threads` on the for loop in the test runner, but was getting strange errors: `Reason: LoadError: package StockFlow did not define the expected module StockFlow, check for typos in package module name`

### CODE REVIEW

File: examples/full_fledged_schema_examples/CausalLoopDiagrams/create_a_causalLoopDiagram.ipynb
Comment (neonWhiteout) [inline examples/full_fledged_schema_examples/CausalLoopDiagrams/create_a_causalLoopDiagram.ipynb:1]: Is there any particular reason you changed it to a Vector?  I'd think it'd work either way.

File: examples/full_fledged_schema_examples_new/stratification/sir_linear_stratification.ipynb [hunk 1]
```diff
@@ -663,7 +661,7 @@
     "    \n",
     "    :flows\n",
     "    CLOUD => f_NewBorn(v_NewBorn) => NormalWeight\n",
-    "    NormalWeight => f_DeathNormalWeight(v_DeathNormalWeight) => ClOUD\n",
+    "    NormalWeight => f_DeathNormalWeight(v_DeathNormalWeight) => CLOUD\n",
     "    NormalWeight => f_BecomingOverWeight(v_BecomingOverWeight) => OverWeight\n",
```
Comment (neonWhiteout) [inline examples/full_fledged_schema_examples_new/stratification/sir_linear_stratification.ipynb:665]: I swear, I thought we've fixed this same thing 3 times over...

**Review by neonWhiteout — APPROVED** [review]
LGTM, thanks Alex.

File: examples/full_fledged_schema_examples/CausalLoopDiagrams/create_a_causalLoopDiagram.ipynb
Comment (Saityi) [reply examples/full_fledged_schema_examples/CausalLoopDiagrams/create_a_causalLoopDiagram.ipynb:1]: It doesn't work either way.

``` 
MethodError: no method matching CausalLoop(::Tuple{Symbol, Symbol, Symbol}, ::NTuple{4, Pair{Symbol, Symbol}})

Stacktrace:
 [1] top-level scope
   @ In[2]:2
```

File: examples/full_fledged_schema_examples_new/stratification/sir_linear_stratification.ipynb (hunk 1 above)
Comment (Saityi) [reply examples/full_fledged_schema_examples_new/stratification/sir_linear_stratification.ipynb:665]: Now that we have CI on these, hopefully it never slips back in! 😃

**Comment by KevinDCarlson** [comment]
Just a remark that the “official” way to accomplish what you’re doing here is with Literate.jl, which will generate Documenter files, notebooks, and scripts from the same markdown source, if you find that shell script inconvenient to maintain at some point.

File: examples/full_fledged_schema_examples/CausalLoopDiagrams/create_a_causalLoopDiagram.ipynb
Comment (neonWhiteout) [reply examples/full_fledged_schema_examples/CausalLoopDiagrams/create_a_causalLoopDiagram.ipynb:1]: Oh dang.  Sorry.

**Pull request merged**